Sources/QuranAudioKit/Preferences/AudioPreferences.swift

```diff
@@ -7,19 +7,50 @@
 
 import Combine
 import Foundation
+import OrderedCollections
 import Preferences
 
 @available(iOS 13.0, *)
 public class AudioPreferences {
     public static let shared = AudioPreferences()
+
     private init() {}
 
     private static let lastSelectedReciterId = PreferenceKey<Int>(key: "LastSelectedQariId", defaultValue: 41)
     private static let audioEndKey = PreferenceKey<Int>(key: "audioEndKey", defaultValue: AudioEnd.juz.rawValue)
+    private static let recentReciterIds = PreferenceKey<[Int]>(key: "recentRecitersIdsKey", defaultValue: [])
+    private static let recentReciterIdsTransfomer = PreferenceTransformer<[Int], OrderedSet<Int>>(
+        rawToValue: { OrderedSet($0) },
+        valueToRaw: { Array($0) }
+    )
+    private static let maxNumOfRecentReciters = 3
 
     @TransformedPreference(audioEndKey, transformer: .rawRepresentable(defaultValue: .juz))
     public var audioEnd: AudioEnd
 
     @Preference(lastSelectedReciterId)
     public var lastSelectedReciterId: Int
+
+    @TransformedPreference(recentReciterIds, transformer: recentReciterIdsTransfomer)
+    public var recentReciterIds: OrderedSet<Int>
+
+    public func recentReciters(_ allReciters: [Reciter]) -> [Reciter] {
+        var recentReciters: [Reciter] = []
+        for recentReciterId in self.recentReciterIds {
+            recentReciters.append(allReciters.filter { $0.id == recentReciterId }[0])
+        }
+        recentReciters.reverse()
+        return recentReciters
+    }
+
+    public func updateRecentRecitersList(_ reciter: Reciter) {
+        // Remove from the set if it exists in the set so it can go to the front (recently selected)
+        recentReciterIds.remove(reciter.id)
+
+        // Remove the least recently selected reciter if the list contains max reciters
+        if recentReciterIds.count >= Self.maxNumOfRecentReciters {
```

Comment:
`Self.maxNumOfRecentReciters` is just a constant that's initialized at the top of the class to 3. According to the [Swift docs](https://docs.swift.org/swift-book/LanguageGuide/TheBasics.html#//apple_ref/doc/uid/TP40014097-CH5-XID_411), Int is preferred:

> Use UInt only when you specifically need an unsigned integer type with the same size as the platform’s native word size. If this isn’t the case, Int is preferred, even when the values to be stored are known to be nonnegative. A consistent use of Int for integer values aids code interoperability, avoids the need to convert between different number types, and matches integer type inference, as described in [Type Safety and Type Inference](https://docs.swift.org/swift-book/LanguageGuide/TheBasics.html#ID322).

Reply:
since this app is completely in swift, we don't need to worry about interoperability. Regardless, we can leave it Int, but it's safer to check that count is greater than 0 before trying to remove at that index.

Reply:
alternatively, make an extension of the Array class to add methods that safely result in no-op for bad input and don't throw errors that can crash the app in prod

Reply:
Added a `!recentReciterIds.isEmpty` condition